File: services/api-gateway/internal/handler/health.go
package handler

import (
	"net/http"

	"github.com/gin-gonic/gin"
)

type healthResponse struct {
	Status  string `json:"status"`
	Service string `json:"service"`
	Version string `json:"version"`
}

func Health(c *gin.Context) {
	c.JSON(http.StatusOK, healthResponse{
		Status:  "ok",
		Service: "api-gateway",
		Version: "0.1.0",
	})
}

func Readiness(c *gin.Context) {
	// TODO: Check downstream services (Redis, RabbitMQ, SIR Core)
	c.JSON(http.StatusOK, gin.H{"ready": true})
}

File: services/sir-core/src/health.rs
use axum::{http::StatusCode, Json};
use serde::Serialize;

#[derive(Serialize)]
pub struct HealthResponse {
    pub status: &'static str,
    pub service: &'static str,
    pub version: &'static str,
}

pub async fn health_check() -> Json<HealthResponse> {
    Json(HealthResponse {
        status: "ok",
        service: "sir-core",
        version: env!("CARGO_PKG_VERSION"),
    })
}

pub async fn readiness_check() -> StatusCode {
    // TODO: Check Redis + RabbitMQ connectivity
    StatusCode::OK
}
